Corrected plugin descriptor name to match config name

Infinitay opened the pull request.

Infinitay commented:
My reason for changing from `onGameStateChanged` to `onGameTick` is primarily to support Robin's daily task. When using `onGameStateChanged`, it wasn't returning the right value of the diaries (returned 0 despite being completed, and returned 1 after I clicked play and was _fully_ loaded into the game)

I hope this doesn't affect the clients performance while running the plugin because of using `onGameTick`. Additionally, if someone knows of a more efficient way to implement `checkCanCollectEcto` method, I'd love to hear how, unless you can push it yourself.

Lastly, I submitted an issue #1157 and if it gets fixed, I will switch over to `onVarbitChanged` if all goes well and my idea on how to handle it efficiently works.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java:
```diff
@@ -99,29 +98,48 @@ else if (event.getKey().equals("showEssence"))
 
 	@Subscribe
-	public void onGameStateChanged(GameStateChanged event)
+	public void onGameTick(GameTick event)
```
Since these are all varbit related, why not use the varbit change event?

Infinitay replied:
Oh I forgot to mention this, sorry. I don't use the varb change event because then all the check methods would return true because all the values are 0.

Adam- replied:
Store the old value and then check the new value to see if it has changed.

sethtroll commented on runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java:
```diff
@@ -99,48 +100,91 @@ else if (event.getKey().equals("showEssence"))
+		return prevStavesVarbVal == value ? (prevStavesVarbVal = value) == 0 : false;
 	}
 
 	private boolean checkCanCollectEssence()
```
check for ardougne medium diary

sethtroll commented on runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java:
```diff
@@ -99,48 +100,91 @@ else if (event.getKey().equals("showEssence"))
+
 	}
 
 	private boolean checkCanCollectStaves()
```
check for varrock easy diary

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java:
```diff
@@ -61,7 +63,10 @@
+	private boolean hasSentHerbMsg, hasSentStavesMsg, hasSentEssenceMsg, hasSentSandMsg, hasSentEctoMsg;
+	private int prevHerbVarbVal, prevStavesVarbVal, prevEssVarbVal, prevSandVarbVal, prevEctoVarbVal;
+
+	private final Set<Integer> MEMBER_WORLDS = Sets.newHashSet(1, 8, 16, 26, 35, 81, 82, 83, 84, 85, 93, 94, 117);
```
Is this variable named correctly? This set contains the f2p worlds, but not the members worlds. `s/MEMBER/F2P/`?

Infinitay replied:
Good catch, my fault, pushing the changed variable now

Kamielvf commented on runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java:
```diff
@@ -99,48 +104,119 @@ else if (event.getKey().equals("showEssence"))
+
+	private boolean checkCanCollectSand()
+	{
+		if (client.getSetting(Varbits.QUEST_A_HAND_IN_THE_SAND) == 160 && !FREE_TO_PLAY_WORLDS.contains(client.getWorld()))
```
Completing the ardy elite diaries makes buckets of sand be automatically delivered to your bank, so this notification has no use at that point.

Infinitay replied:
Fair point, I'll remove this later tonight or tomorrow

Infinitay commented:
Rebased and updated. If someone could test this out before I can, which would be by the end of the week that would be wonderful. I did however test it out in F2P, but that doesn't say much seeing as I can't claim the dailies. 

Nonetheless, I didn't change much of the logic that was committed prior to my latest two commits, so everything in theory should be working.

- Moved logic into a new method
- Added `Client#isMember` to API (idk why it wasn't already)
- Fixed ecto daily (was returning true when the varbit value was 0, meaning I can claim it)
- Initialized all previous values to 0 (0 = we can claim)
- Rebased to upstream

Infinitay commented:
Rebased and pushed new changes

- Added _documentation_ to the check methods
- Added flax to bowstring daily task
- Added random runes daily task
- Added capability to reset sent messages (allows sending notifications if switching accounts when the message was already sent)

Infinitay commented:
This is hopefully the final commit. Here's a list of notable things I've changed with the latest commits:

- Fixed ecto daily not sending message when collected more than one, but less than the daily specified amount
- Changed how resetting notification upon account change is handled
- Added new `resetVariables` method to reset pre-specified variables
- Changed how config change is handled
